daemon/logger/loggerutils/logfile.go

```diff
@@ -224,6 +364,25 @@ func (w *LogFile) openRotatedFiles() (files []*os.File, err error) {
 			if !os.IsNotExist(err) {
 				return nil, err
 			}
+
+			fileName := fmt.Sprintf("%s.%d.gz", w.f.Name(), i-1)
```

Comment:
`w.f.Name()` may be racey here. We need to grab the rotation lock at the top of this function, I think.